src/Utils/ChatDataSharing.cs

```diff
@@ -0,0 +1,109 @@
+﻿using ProtoBuf;
+using System;
+using System.IO.Compression;
+using System.IO;
+using Vintagestory.API.Client;
+using Vintagestory.API.Common;
+using ProspectorInfo.Map;
+using System.Linq;
+using System.Text;
+
+namespace ProspectorInfo.Utils
+{
+    internal class ChatDataSharing
+    {
+        private static readonly string CHAT_PREFIX = "ProspectingData;";
+
+        private readonly ICoreClientAPI api;
+        private readonly ProspectorOverlayLayer prospectorOverlayLayer;
+        public ChatDataSharing(ICoreClientAPI api, ProspectorOverlayLayer prospectorOverlayLayer)
+        {
+            this.api = api;
+            this.prospectorOverlayLayer = prospectorOverlayLayer;
+            api.Event.ChatMessage += OnChatMessage;
+        }
+
+        private void OnChatMessage(int groupId, string message, EnumChatType chattype, string data) {
+            if (chattype != EnumChatType.OthersMessage) 
+                return;
+
+            int startIdx = message.IndexOf(CHAT_PREFIX);
+            if (startIdx == -1)
+                return;
+
+            var result = DeserializeFromBase64<ProspectorMessages>(message.Substring(startIdx + CHAT_PREFIX.Length));
+            if (result != null)
+                prospectorOverlayLayer.AddOrUpdateProspectingData(result.Values.ToArray());
+        }
+
+        public void ShareData(ProspectorMessages messages)
+        {
+            string data = SerializeToBase64<ProspectorMessages>(messages);
+            if (data == null) 
+            {
+                return;
+            }
+            // Add some whitespace before sending, to avoid lag spikes when rendering the chat message.
+            // The base64 deserializer ignores whitespace.
+            api.SendChatMessage(CHAT_PREFIX + AddWhitespace(data, 64));
```

Comment:
Have you notified @tyronx about the issue ?

Reply:
Not yet. Tbh, I'm a bit ashamed of this solution, as the chat is clearly not intended for such a use case. 

I guess a better solution would be to change the client api to add some kind of broadcast, where the server just relays a packet to all clients. Clients that recognize the packet, i.e., have a certain handler from a mod, can then process the packet. I will open an issue when I find the time.

Reply:
I am aware of the text performance limits of the chat window, yes